src/app/institutional/pages/institutional/institutional.component.ts

```diff
@@ -0,0 +1,185 @@
+          ])
+          this.clear()
+        },
+        (error) => {
+          console.log('Error getting disco feed' + JSON.stringify(error))
+        }
+      )
+  }
+
+  ngOnInit() {}
+
+  onSubmit() {
+    if (this.institutitonalForm.valid) {
+      const dateCookie = new Date(
+        new Date().getTime() + this.samlIdPCookieTTL * 24 * 60 * 60 * 1000
+      )
+      this._cookie.set(
+        '_saml_idp',
+        this.getCookieSaml(this.entityID),
+        dateCookie !== null ? dateCookie : this.samlIdPCookieTTL
+      )
+
+      const defaultReturn =
+        'https:' +
+        environment.BASE_URL +
+        'Shibboleth.sso/Login?SAMLDS=1&target=' +
+        encodeURIComponent(
+          'https:' + environment.BASE_URL + 'shibboleth/signin'
+        )
+
+      this.navigateTo(
+        defaultReturn + '&entityID=' + encodeURIComponent(this.entityID)
+      )
+    }
+  }
+
+  private _filter(value: string): string[] {
+    if (value === '' || value.length >= 3) {
+      const filterValue = value.toLowerCase()
+
+      if (value === '' && this.retrieveAllFiltered) {
+        return this.retrieveAllFiltered
+      } else if (value === '') {
+        this.retrieveAllFiltered = this.options
+          .filter((institution) =>
+            institution.DisplayNames.some((displayNames) =>
+              displayNames.value.toLowerCase().includes(filterValue)
+            )
+          )
+          .map((result) => {
+            return result.DisplayNames.filter(
+              (subElement) => subElement.lang === 'en'
+            ).map((en) => {
+              return en.value
+            })
+          })
+        return this.retrieveAllFiltered
+      } else {
+        return this.options
+          .filter((institution) =>
+            institution.DisplayNames.some((displayNames) =>
+              displayNames.value.toLowerCase().includes(filterValue)
+            )
+          )
+          .slice(0, 20)
+          .map((result) => {
+            return result.DisplayNames.filter(
+              (subElement) => subElement.lang === 'en'
+            ).map((en) => {
+              return en.value
+            })
+          })
+      }
+    }
+  }
+
+  clear() {
+    this.logoInstitution = undefined
+    this.institutitonalForm.controls['institution'].setValue('')
+    this.filteredOptions = this.institutionFormControl.valueChanges.pipe(
+      startWith(''),
+      map((value) => this._filter(value))
+    )
+  }
+
+  selected(institutionSelected) {
+    this.logoInstitution = undefined
+    this.institution = this.options.filter((institution) =>
+      institution.DisplayNames.some(
+        (displayNames) =>
+          displayNames.value.toLowerCase() ===
+          institutionSelected[0].toLowerCase()
+      )
+    )[0]
+    this.entityID = this.institution.entityID
+    if (this.institution.Logos) {
+      this.logoInstitution = this.institution.Logos[0].value
+    }
+  }
+
+  getCookieSaml(entityId) {
```

Comment:
@leomendoza123 Is it from shibboleth https://github.com/ORCID/ORCID-Source/blob/master/orcid-web/src/main/webapp/static/javascript/shibboleth-embedded-ds/1.1.0/idpselect.js

Reply:
It this cookie required? it seems to me that it is only used by the old autocomplete to remember what institution has the user used before...

But if you check and is required for the institutional login process... please let's use the source code for this, instead of hard to read minified code. 

You can download the source code from https://shibboleth.net/downloads/embedded-discovery-service/latest/shibboleth-embedded-ds-1.2.2.tar.gz 

You will find this function called `saveUserSelectedIdPs` on `nonminimized/idpselect.js` witch build a `cookieData` array and makes a string with `cookieData.join('%20')` which seems to be the equivalent for this. Please also add a comment to explain where this code is coming from as well (shibboleth-embedded-1.2.2).

Reply:
Feel free to create an issue if you want to solve this as part of the institutional linking

Reply:
Good catch!. I only checked the minified code and I didn't see that function and It looks that you are right and the cookie doesn't affect the login process apparently. I will remove it then.